SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -590,3 +590,16 @@ theorem Bool.xor_decide (p q : Prop) [dp : Decidable p] [Decidable q] :
     (decide p).xor (decide q) = decide (p ≠ q) := by
   cases' dp with pt pt
   <;> simp [pt]
+
+theorem Bool.xor_shift {a b c : Bool} : xor a b = c ↔ b = xor a c := by
+  cases a
```

Comment:
Why is this named shift? Is there precedence somewhere in the lean codebase and/or mathlib?

Reply:
I'd call it `xor_inv_left`, because it's saying that $$a \oplus b = c$$ iff $$a \oplus a \oplus b = a \oplus c$$, which implies $$b = a \oplus c$$ (modulo 2,  $$a = a^{-1}$$)